### ISSUE TITLE
GPU-memory not freed after "Classify Many". 

### ISSUE THREAD

**Issue by riteshpradhan** [NVIDIA/DIGITS#679]
I tried to do "classify many" with just 10 test images to check my GPU status but it failed with 

> "Error: Unable to connect to NVIDIA DIGITS"

 error.  

content of

>  /var/log/digits/digits.log

 is 

> 2016-04-13 18:25:48 [1281] [CRITICAL] WORKER TIMEOUT (pid:3237)
> 2016-04-13 18:25:48 [1281] [CRITICAL] WORKER TIMEOUT (pid:3237)
> 2016-04-13 18:25:49 [3308] [INFO] Booting worker with pid: 3308

During the process, What I see is that my GPU memory is not being freed once it fails. 

```
> Wed Apr 13 18:34:50 2016
> +------------------------------------------------------+
> | NVIDIA-SMI 352.79     Driver Version: 352.79         |
> |-------------------------------+----------------------+----------------------+
> | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
> | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
> |===============================+======================+======================|
> |   0  GRID K520           Off  | 0000:00:03.0     Off |                  N/A |
> | N/A   44C    P0    42W / 125W |    531MiB /  4095MiB |      0%      Default |
> +-------------------------------+----------------------+----------------------+
> 
> +-----------------------------------------------------------------------------+
> | Processes:                                                       GPU Memory |
> |  GPU       PID  Type  Process name                               Usage      |
> |=============================================================================|
> |    0      1281    C   /usr/bin/python                                 36MiB |
> +-----------------------------------------------------------------------------+
```

My 531MB of memory is being consumed by a failed process. 

Also If the process succeeded it will not free those used memory. 

```
Wed Apr 13 19:00:30 2016
+------------------------------------------------------+
| NVIDIA-SMI 352.79     Driver Version: 352.79         |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GRID K520           Off  | 0000:00:03.0     Off |                  N/A |
| N/A   43C    P0    42W / 125W |    971MiB /  4095MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0      1281    C   /usr/bin/python                                 36MiB |
|    0      3308    C   /usr/bin/python                                438MiB |
+-----------------------------------------------------------------------------+


```

**Comment by gheinrich** [comment]
Hi @riteshpradhan the memory utilization that you see even when there are no processes running is reserved by the driver. As per the nvidia-smi [doc](http://developer.download.nvidia.com/compute/cuda/6_0/rel/gdk/nvidia-smi.331.38.pdf):

> FB Memory Usage
> The driver may also reserve a small amount of memory for internal use, even without active work on the GPU.

**Comment by gheinrich** [comment]
Hopefully your question is answered. Closing the bug for now. Thanks for the report!

**Comment by riteshpradhan** [comment]
So do you mean, GPU drive reserve memory and that keeps on increasing for every failed "classify_many" operation? That's what happened in my case.

**Comment by gheinrich** [comment]
Re-opening.

Your initial report doesn't provide evidence of an increasing GPU allocated memory after every failed `classify_many` operation. If that is the case, can you do this 10 times and provide the relevant lines of `nvidia-smi` output? Also, please tells us which version of DIGITS you are using.

**Comment by riteshpradhan** [comment]
Previous State:
<img width="648" alt="screen shot 2016-04-18 at 11 45 47 am" src="https://cloud.githubusercontent.com/assets/2297214/14612358/40d94caa-055c-11e6-9fab-c543b594602a.png">

Classify Many Failed State: (Job with PID: 3308 killed (total previous usage was 951MB) 
<img width="646" alt="screen shot 2016-04-18 at 11 46 38 am" src="https://cloud.githubusercontent.com/assets/2297214/14612376/54bbde0e-055c-11e6-95fd-7e0ff97de190.png">

Success: (New Process)
<img width="638" alt="screen shot 2016-04-18 at 11 46 56 am" src="https://cloud.githubusercontent.com/assets/2297214/14612427/8fa8a6f0-055c-11e6-8098-9fb0d189795d.png">


Failed:
<img width="644" alt="screen shot 2016-04-18 at 11 47 09 am" src="https://cloud.githubusercontent.com/assets/2297214/14612437/9c18cdf2-055c-11e6-82b0-89d20b77442f.png">

Success (running):
<img width="634" alt="screen shot 2016-04-18 at 11 47 26 am" src="https://cloud.githubusercontent.com/assets/2297214/14612439/9c191230-055c-11e6-95cf-875b85594dad.png">

Success (completion):
<img width="643" alt="screen shot 2016-04-18 at 11 47 40 am" src="https://cloud.githubusercontent.com/assets/2297214/14612441/9c1a434e-055c-11e6-9fb4-9b1223182f4c.png">

Failed:
<img width="645" alt="screen shot 2016-04-18 at 11 48 16 am" src="https://cloud.githubusercontent.com/assets/2297214/14612436/9c18c5d2-055c-11e6-9d4e-e56297a0e328.png">

Success:
<img width="638" alt="screen shot 2016-04-18 at 11 48 51 am" src="https://cloud.githubusercontent.com/assets/2297214/14612440/9c194a70-055c-11e6-9a4f-c2a7c4b9b23f.png">

Failed:
<img width="644" alt="screen shot 2016-04-18 at 11 51 36 am" src="https://cloud.githubusercontent.com/assets/2297214/14612438/9c18ddf6-055c-11e6-914a-2d197b091205.png">

Success:
<img width="646" alt="screen shot 2016-04-18 at 11 52 25 am" src="https://cloud.githubusercontent.com/assets/2297214/14612443/9c20660c-055c-11e6-98a0-29c36a927070.png">

For every failed classify_many I get this error in web.
<img width="822" alt="screen shot 2016-04-18 at 11 51 47 am" src="https://cloud.githubusercontent.com/assets/2297214/14612442/9c202e26-055c-11e6-925e-a1b7c50dacbe.png">

**Comment by lukeyeager** [comment]
That looks bad.

Try running with the `master` branch. This may have been fixed with https://github.com/NVIDIA/DIGITS/pull/573.

**Comment by gheinrich** [comment]
Closing due to inactivity - assumed fixed with #573
